src/operator/src/statement/set.rs

```diff
@@ -0,0 +1,179 @@
+            }
+            .fail();
+        }
+    };
+    // For the sake of simplicity, we only support "UTF8" ("UNICODE" is the alias for it,
+    // see https://www.postgresql.org/docs/current/multibyte.html#MULTIBYTE-CHARSET-SUPPORTED).
+    // "UTF8" is universal and sufficient for almost all cases.
+    // GreptimeDB itself is always using "UTF8" as the internal encoding.
+    ensure!(
+        encoding == "UTF8" || encoding == "UNICODE",
+        NotSupportedSnafu {
+            feat: format!("client encoding of '{}'", encoding)
+        }
+    );
+    Ok(())
+}
+
+// return Ok(Some(new_val)) if value not exist or new_value equals to value
+// else return error.
+// This is only used for handling datestyle
+fn merge_datestyle_value<T>(value: Option<T>, new_value: Option<T>) -> Result<Option<T>>
+where
+    T: PartialEq,
+{
+    match (&value, &new_value) {
+        (None, _) => Ok(new_value),
+        (_, None) => Ok(value),
+        (Some(v1), Some(v2)) if v1 == v2 => Ok(new_value),
+        _ => InvalidSqlSnafu {
+            err_msg: "Conflicting \"datestyle\" specifications.",
+        }
+        .fail(),
+    }
+}
+
+fn try_parse_datestyle(expr: &Expr) -> Result<(Option<PGDateTimeStyle>, Option<PGDateOrder>)> {
+    enum ParsedDateStyle {
+        Order(PGDateOrder),
+        Style(PGDateTimeStyle),
+    }
+    fn try_parse_str(s: &str) -> Result<ParsedDateStyle> {
+        PGDateTimeStyle::try_from(s)
+            .map_or_else(
+                |_| PGDateOrder::try_from(s).map(ParsedDateStyle::Order),
+                |style| Ok(ParsedDateStyle::Style(style)),
+            )
+            .context(InvalidConfigValueSnafu)
+    }
+    match expr {
+        Expr::Identifier(Ident {
+            value: s,
+            quote_style: _,
+        })
+        | Expr::Value(Value::SingleQuotedString(s))
+        | Expr::Value(Value::DoubleQuotedString(s)) => {
+            s.split(',')
+                .map(|s| s.trim())
+                .try_fold((None, None), |(style, order), s| match try_parse_str(s)? {
+                    ParsedDateStyle::Order(o) => {
+                        Ok((style, merge_datestyle_value(order, Some(o))?))
+                    }
+                    ParsedDateStyle::Style(s) => {
+                        Ok((merge_datestyle_value(style, Some(s))?, order))
+                    }
+                })
+        }
+        _ => NotSupportedSnafu {
+            feat: "Not supported expression for datestyle",
+        }
+        .fail(),
+    }
+}
+
+pub fn set_datestyle(exprs: Vec<Expr>, ctx: QueryContextRef) -> Result<()> {
+    // ORDER,
+    // STYLE,
+    // ORDER,ORDER
+    // ORDER,STYLE
+    // STYLE,ORDER
+    let (style, order) = exprs
+        .iter()
+        .try_fold((None, None), |(style, order), expr| {
+            let (new_style, new_order) = try_parse_datestyle(expr)?;
+            Ok((
+                merge_datestyle_value(style, new_style)?,
+                merge_datestyle_value(order, new_order)?,
```

Comment:
Can you provide an example or test case for this, I didn't quite get the logic here.

Reply:
I've add some related test. Postgresql support quiet complicated input here.